scripts/src/javatests/org/oppia/android/scripts/testfile/TestFileCheckTest.kt

```diff
@@ -0,0 +1,98 @@
+package org.oppia.android.scripts.testfile
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [TestFileCheck]. */
+class TestFileCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+  private val TEST_FILE_CHECK_PASSED_OUTPUT_INDICATOR = "TEST FILE CHECK PASSED"
+  private val TEST_FILE_CHECK_FAILED_OUTPUT_INDICATOR = "TEST FILE CHECK FAILED"
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testTestFileCheck_prodFileWithCorrespondingTestFile_testFileIsPresent() {
+    tempFolder.newFile("testfiles/ProdFile.kt")
+    tempFolder.newFile("testfiles/ProdFileTest.kt")
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(TEST_FILE_CHECK_PASSED_OUTPUT_INDICATOR)
+  }
+
+  @Test
+  fun testTestFileCheck_missTestFile_testFileIsNotPresent() {
+    tempFolder.newFile("testfiles/ProdFile1.kt")
+    tempFolder.newFile("testfiles/ProdFile1Test.kt")
+    tempFolder.newFile("testfiles/ProdFile2.kt")
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(TEST_FILE_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString()).contains("${retrieveTestFilesDirectoryPath()}/ProdFile2.kt")
+  }
+
+  @Test
+  fun testTestFileCheck_missTestFilesForMultipleProdFiles_testFileIsNotPresent() {
+    tempFolder.newFile("testfiles/ProdFile1.kt")
+    tempFolder.newFile("testfiles/ProdFile1Test.kt")
+    tempFolder.newFile("testfiles/ProdFile2.kt")
+    tempFolder.newFile("testfiles/ProdFile3.kt")
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(TEST_FILE_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString()).contains("${retrieveTestFilesDirectoryPath()}/ProdFile2.kt")
```

Comment:
Similar to earlier tests, we should actually validate that we're providing sufficient context & verifying the script's formulated error messages. Those are specifically designed to convey information in a certain way & it's important to continue that.

It's less important to verify the output of downstream libraries that produce errors for us (such as for the XML check), but the errors we output should be validated.

Ditto for earlier PRs if applicable.

Reply:
Apologies! This is now rectified and updated. PTAL.